### PR TITLE
Randomize host order during uploads and downloads

### PR DESCRIPTION
**Pull request by ChrisSchinnerl**

### CODE REVIEW

File: worker/transfer.go [hunk 1]
```diff
@@ -32,6 +33,10 @@ func parallelUploadSlab(ctx context.Context, shards [][]byte, hosts []sectorStor
 		return nil, fmt.Errorf("fewer hosts than shards, %v<%v", len(hosts), len(shards))
 	}
 
+	// Randomize order of hosts to make sure we don't always start with the
+	// same one.
+	frand.Shuffle(len(hosts), func(i, j int) { hosts[i], hosts[j] = hosts[j], hosts[i] })
```
Comment (peterjan) [inline worker/transfer.go:38]: I would've shuffled on the caller side. I think `parallelUploadSlab` and `parallelDownloadSlab` should still respect the order of the given hosts slice for the time being.

**Review by peterjan — APPROVED** [review]

File: worker/transfer.go (hunk 1 above)
Comment (peterjan) [reply worker/transfer.go:38]: I think by leaving it up to the caller, the upload/download code is slightly more generic and reusable. I might want to use a different upload/download strategy for different use cases. I guess you might be able to pass in a `strategy` but I don't see why you would have that code in the upload/download code.

Comment (ChrisSchinnerl) [reply worker/transfer.go:38]: Keeping it lower level for now until we have the need for multiple ordering options

Comment (peterjan) [reply worker/transfer.go:38]: Good stuff 👍

**Pull request merged**